src/Features/VisualBasic/Portable/CodeRefactorings/InlineMethod/VisualBasicInlineMethodRefactoringProvider.vb

```diff
@@ -0,0 +1,71 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Composition
+Imports Microsoft.CodeAnalysis.CodeRefactorings
+Imports Microsoft.CodeAnalysis.Host.Mef
+Imports Microsoft.CodeAnalysis.InlineMethod
+Imports Microsoft.CodeAnalysis.VisualBasic.LanguageServices
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.CodeRefactorings.InlineTemporary
+
+    <ExportCodeRefactoringProvider(LanguageNames.VisualBasic, Name:=NameOf(PredefinedCodeRefactoringProviderNames.InlineMethod)), [Shared]>
+    <Export(GetType(VisualBasicInlineMethodRefactoringProvider))>
+    Friend Class VisualBasicInlineMethodRefactoringProvider
+        Inherits AbstractInlineMethodRefactoringProvider(Of MethodBlockSyntax, StatementSyntax, ExpressionSyntax, InvocationExpressionSyntax)
+
+        <ImportingConstructor>
+        <Obsolete(MefConstruction.ImportingConstructorMessage, True)>
+        Public Sub New()
+            MyBase.New(VisualBasicSyntaxFacts.Instance, VisualBasicSemanticFactsService.Instance)
+        End Sub
+
+        Protected Overrides Function GetInlineExpression(methodBlock As MethodBlockSyntax) As ExpressionSyntax
+            Dim statements = methodBlock.Statements
+            If statements.Count = 1 Then
+                Dim singleStatement = statements(0)
+                Dim returnStatement = TryCast(singleStatement, ReturnStatementSyntax)
+                If returnStatement IsNot Nothing Then
+                    Return returnStatement.Expression
+                End If
+
+                Dim expressionStatement = TryCast(singleStatement, ExpressionStatementSyntax)
+                If expressionStatement IsNot Nothing Then
+                    Return expressionStatement.Expression
+                End If
+
+                Dim throwStatement = TryCast(singleStatement, ThrowStatementSyntax)
+                If throwStatement IsNot Nothing Then
+                    Return throwStatement.Expression
+                End If
+            End If
+            Return Nothing
+        End Function
+
+        Protected Overrides Function GetEnclosingMethodLikeNode(syntaxNode As SyntaxNode) As SyntaxNode
+            While syntaxNode IsNot Nothing
+                If TypeOf syntaxNode Is MethodBlockBaseSyntax OrElse TypeOf syntaxNode Is LambdaExpressionSyntax Then
```

Comment:
do you have tests for single-line vs multiline vb lambdas?

Reply:
I think I have Multi-line Lambda test case. But no single-line test. Will add taht